hooks/tk-3dsmaxplus_actions.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sgtk
+import MaxPlus
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class MaxActions(HookBaseClass):
+    """
+    Shotgun Panel Actions for 3dsMax
+    """
+    
+    def generate_actions(self, sg_data, actions, ui_area):
+        """
+        Returns a list of action instances for a particular object.
+        The data returned from this hook will be used to populate the 
+        actions menu.
+    
+        The mapping between Shotgun objects and actions are kept in a different place
+        (in the configuration) so at the point when this hook is called, the app
+        has already established *which* actions are appropriate for this object.
+        
+        This method needs to return detailed data for those actions, in the form of a list
+        of dictionaries, each with name, params, caption and description keys.
+        
+        The ui_area parameter is a string and indicates where the item is to be shown. 
+        
+        - If it will be shown in the main browsing area, "main" is passed. 
+        - If it will be shown in the details area, "details" is passed.
+                
+        :param sg_data: Shotgun data dictionary with all the standard shotgun fields.
```

Comment:
nitpick: `:param dict sg_data:` to specify argument type.

Same for the others.